src/api/Buffer.ts

```diff
@@ -243,20 +255,22 @@ export class Buffer extends BaseApi {
    */
   listen(eventName: string, cb: Function): Function {
     if (!this.isAttached) {
-      this[ATTACH]();
-      this.isAttached = true;
+      this[ATTACH]().then(attached => {
```

Comment:
`await` instead of `then` here too (will also have to make this method `async`)

Reply:
I'm fine with `async`, but that would be break change.

Reply:
Ah you're right, I didn't read down far enough, let's revert this change. Sorry about that.